src/encoded/visualization.py

```diff
@@ -19,15 +26,196 @@
+                        "missing"   : TERM_NAME_FOR_NO_VALUE,
+                        "size"      : 30
+                    }
+                }
+                for group_by_field in group_by_fields if group_by_field is not None
+            }
+        else:
+            histogram_sub_aggs = None
+
+    # Create an agg item for each interval in `date_histogram_intervals` x each date field in `date_histogram_fields`
+    # TODO: Figure out if we want to align these up instead of do each combination.
+    outer_date_histogram_agg = {}
+    for interval in date_histogram_intervals:
+        for dh_field in date_histogram_fields:
+            outer_date_histogram_agg[interval + '_interval_' + dh_field] = {
+                "date_histogram" : {
+                    "field": "embedded." + dh_field,
+                    "interval": interval_to_es_interval[interval],
+                    "format": "yyyy-MM-dd"
+                }
+            }
+            if histogram_sub_aggs:
+                outer_date_histogram_agg[interval + '_interval_' + dh_field]['aggs'] = histogram_sub_aggs
+
+    search_param_lists['limit'] = search_param_lists['from'] = [0]
+    subreq          = make_search_subreq(request, '{}?{}'.format('/search/', urlencode(search_param_lists, True)) )
+    search_result   = perform_search_request(None, subreq, custom_aggregations=outer_date_histogram_agg)
+
+    # remove unnecessary fields from result
+    for field_to_delete in ['@context', '@id', '@type', '@graph', 'title', 'filters', 'facets', 'sort', 'clear_filters', 'actions', 'columns']:
+        if search_result.get(field_to_delete) is None:
+            continue
+        del search_result[field_to_delete]
+
+    return search_result
+
+
+def convert_date_range(date_range_str):
+    
+    data_range_split = date_range_str.split('|')
+    preset = data_range_split[0]
+    if preset not in DATE_RANGE_PRESETS:
+        raise IndexError('"{}" is not one of {}.'.format(preset, ', '.join(DATE_RANGE_PRESETS)))
+    
+    today = datetime.today()
+    first_day_this_month = today.replace(day=1)
+    date_from, date_to = None, None
+
+    if preset == 'thismonth':
+        date_from = first_day_this_month
+    elif preset == 'previousmonth':
+        date_from = first_day_this_month - relativedelta(months=1)
+        date_to = first_day_this_month - relativedelta(days=1)
+    elif preset == 'last3months':
+        date_from = first_day_this_month - relativedelta(months=2)
+    elif preset == 'last6months':
+        date_from = first_day_this_month - relativedelta(months=5)
+    elif preset == 'last12months':
+        date_from = first_day_this_month - relativedelta(months=11)
+    elif preset == 'thisyear':
+        date_from = datetime(today.year, 1, 1)
+    elif preset == 'previousyear':
+        date_from = datetime(today.year - 1, 1, 1)
+        date_to = datetime(today.year - 1, 12, 31)
+    elif preset == 'custom':
```

Comment:
Stylistic comment - instead of doing it this way, you could write a function that builds a dictionary mapping your presets to the ``datetime`` values and just index into the dictionary in this function ie:

```
def generate_preset_table():
    today = datetime.today()
    first_day_this_month = today.replace(day=1)
    date_from, date_to = None, None
    d = {
        'thismonth': (first_day_this_month, None),
        'previousmonth': (first_day_this_month - relativedelta(months=1), first_day_this_month - relativedelta(days=1))
        ...
     }
     return d
```

You can make this even cleaner by using a `NamedTuple` for `date_from` and `date_to`, but a standard tuple will work as well.